src/EnergyPlus/RoomAirModelAirflowNetwork.cc

```diff
@@ -357,24 +360,60 @@ namespace RoomAirModelAirflowNetwork {
                              ++EquipLoop) { // loop over all the equip for a single room air node
                             // Check zone equipment name
                             for (I = 1; I <= state.dataZoneEquip->ZoneEquipList(LoopZone).NumOfEquipTypes; ++I) { // loop over all equip types
-                                if (UtilityRoutines::SameString(
-                                        state.dataZoneEquip->ZoneEquipList(LoopZone).EquipName(I),
-                                        state.dataRoomAirMod->RoomAirflowNetworkZoneInfo(LoopZone).Node(LoopAirNode).HVAC(EquipLoop).Name)) {
-                                    if (state.dataRoomAirMod->RoomAirflowNetworkZoneInfo(LoopZone)
-                                            .Node(LoopAirNode)
-                                            .HVAC(EquipLoop)
-                                            .EquipConfigIndex == 0)
-                                        state.dataRoomAirMod->RoomAirflowNetworkZoneInfo(LoopZone)
-                                            .Node(LoopAirNode)
-                                            .HVAC(EquipLoop)
-                                            .EquipConfigIndex = I;
-                                    EquipFound(I) = true;
-                                    SupplyFrac(I) =
-                                        SupplyFrac(I) +
-                                        state.dataRoomAirMod->RoomAirflowNetworkZoneInfo(LoopZone).Node(LoopAirNode).HVAC(EquipLoop).SupplyFraction;
-                                    ReturnFrac(I) =
-                                        ReturnFrac(I) +
-                                        state.dataRoomAirMod->RoomAirflowNetworkZoneInfo(LoopZone).Node(LoopAirNode).HVAC(EquipLoop).ReturnFraction;
+                                if (state.dataZoneEquip->ZoneEquipList(LoopZone).EquipTypeEnum(I) == DataZoneEquipment::ZoneEquip::AirDistUnit) {
+                                    if (state.dataZoneAirLoopEquipmentManager->GetAirDistUnitsFlag) {
+                                        numAirDistUnits =
+                                            state.dataInputProcessing->inputProcessor->getNumObjectsFound(state, "ZoneHVAC:AirDistributionUnit");
+                                        ZoneAirLoopEquipmentManager::GetZoneAirLoopEquipment(state);
+                                        state.dataZoneAirLoopEquipmentManager->GetAirDistUnitsFlag = false;
+                                    }
+                                    for (int AirDistUnitNum = 1; AirDistUnitNum <= numAirDistUnits; ++AirDistUnitNum) {
```

Comment:
Check air terminal objects under AirDistUnit

Reply:
I could probably propose some cleanups in here, but the arrays and nested structures are quite intertwined, so I'm not actually sure what `auto &` references would be most beneficial.  And this is only done during GetInput so I don't think it's worth rigoring over too much.  If this fixes the defect, I would just let it be.